src/ClientGenerator/ClientGenerator.cs

```diff
@@ -347,11 +347,22 @@ private static bool CheckFilesMatch(FileInfo file1, FileInfo file2)
                     }
                 }
 
-                if (options.ServerGen && !type.IsAbstract && (TypeUtils.IsGrainClass(type) || TypeUtils.IsSystemTargetClass(type)))
+                if (options.ServerGen && !type.IsAbstract && (GrainInterfaceData.IsAddressable(type)))
```

Comment:
Why IAddressable and not IGrain? There are more things that implement IAddressable, such as IGrainObserver, GrainReference.